e2e/tests/snapp_tests.rs

```diff
@@ -0,0 +1,147 @@
+use e2e::common::{wait_for, wait_for_condition, FutureWaitExt};
+use e2e::snapp::{await_state_transition, setup_snapp};
+use ethcontract::web3::api::Web3;
+use ethcontract::web3::transports::Http;
+use ethcontract::web3::types::{H160, H256, U128, U256};
+use ethcontract::Account;
+use std::str::FromStr;
+
+#[test]
+fn snapp_deposit_withdraw() {
```

Comment:
Maybe we can use the snapshotting function to reset the ganache server back to the state before the test.

Reply:
I remember this... Can check if it still works!

Reply:
Actually, I am not sure this will be sufficient here because we also need to remove all contents from the database (i.e. restart the service anyway).